### DISCUSSION TITLE
refactor: Address ESLint warnings

### DISCUSSION THREAD

**Discussion by danorlando** [danny-avila/LibreChat#504] [Help Wanted]
After re-doing our eslint configuration, some issues were exposed. The biggest concern is the exhaustive dependencies requirement for react useEffect, where we found that in some cases, listing all of the dependencies caused infinite loops on dependency updates. This needs to be investigated further and some refactoring will likely be required.

**Comment by Cfomodz** [comment]
@danorlando I believe this discussion is ready to be closed :) 🚀
